File: programas_en_desarrollo/python_projects/satellite_data/obtain_data_from_satellite_data.py
import re
from datetime import datetime

# Read the contents of the file
input_filename = "Track.txt"
with open(input_filename, "r") as file:
    file_content = file.read()

# Replace "a. m." with "AM" and "p. m." with "PM"
file_content = re.sub(r"[a]\. [m]\.", "AM", file_content)
file_content = re.sub(r"[p]\. [m]\.", "PM", file_content)

# Search for date, time, latitude and longitude strings using regular expressions
date_time_pattern = r"(\d{2}/\d{2}/\d{4} \d{2}:\d{2}:\d{2} [AP][M])"
coordinate_pattern = r"N(\d+\.\d+) W(\d+\.\d+)"
date_time_matches = re.findall(date_time_pattern, file_content)
coordinate_matches = re.findall(coordinate_pattern, file_content)

print("\nDate, time, latitude and longitude strings found!")

# Write the results to a new file
output_filename = "clean_satellite_data.txt"
with open(output_filename, "w") as file:
    for i in range(len(date_time_matches)):
        date_time = date_time_matches[i]
        coordinate = coordinate_matches[i]
        latitude = coordinate[0]
        longitude = coordinate[1]
        # Convert 12-hour clock to 24-hour clock
        date_time_24h = datetime.strptime(date_time, "%d/%m/%Y %I:%M:%S %p").strftime("%d/%m/%Y %H:%M:%S")
        file.write(f"{date_time_24h}, {latitude}, {longitude}\n")

File: scripts_descarga_datos/descargar_datos_matlab/get_data_satellite/obtain_data_from_satellite_data.m
%% Read the contents of the file
inputFileName = 'Track.txt'; % Reemplaza con el nombre de tu archivo
fileContent = fileread(inputFileName);

%% Replace "a. m." with "AM" and "p. m." with "PM"
fileContent = regexprep(fileContent, '[a]\. [m]\.', 'AM');
fileContent = regexprep(fileContent, '[p]\. [m]\.', 'PM');

%% Search for date, time, latitude and longitude strings using regular expressions
dateTimePattern = '(\d{2}/\d{2}/\d{4} \d{2}:\d{2}:\d{2} [AP][M])';
coordinatePattern = 'N(\d+\.\d+) W(\d+\.\d+)';

dateTimeMatches = regexp(fileContent, dateTimePattern, 'tokens');
coordinateMatches = regexp(fileContent, coordinatePattern, 'tokens');

%% Write the results to a new file
outputFileName = 'clean_satellite_data.txt';
fid = fopen(outputFileName, 'w');

fprintf('Date, time, latitude and longitude strings found!\n');
for i = 1:length(dateTimeMatches)
    dateTimeTokens = dateTimeMatches{i};

    % Converts 12-hours clock to 24-hour clock
    dateTime24 = string(datetime(dateTimeTokens{1}, 'InputFormat', 'dd/MM/yyyy hh:mm:ss a', 'Format', 'dd/MM/yyyy HH:mm:ss'));

    latitude = coordinateMatches{i}{1};
    longitude = coordinateMatches{i}{2};

    fprintf(fid, '%s, %s, %s\n', dateTime24, latitude, longitude);
end

fclose(fid);